src/registry/docsetregistry.cpp

```diff
@@ -120,25 +124,32 @@ void DocsetRegistry::_addDocset(const QString &path)
     emit docsetAdded(name);
 }
 
-void DocsetRegistry::search(const QString &query)
+void DocsetRegistry::search(const QString &query, CancellationToken token)
 {
-    // Only invalidate queries
-    if (query.isEmpty())
-        return;
-
-    QMetaObject::invokeMethod(this, "_runQuery", Qt::QueuedConnection, Q_ARG(QString, query));
+    qRegisterMetaType<CancellationToken>("CancellationToken");
```

Comment:
I remember that I had issues when registering the meta type here earlier.
However, I no longer recall the exact problem and after removing one of the calls did not break zeal.